src/workers/notifications/worker.ts

```diff
@@ -11,12 +12,48 @@ export type NotificationHandlerReturn =
     }[]
   | undefined;
 
+export type NotificationWorkerHandler = (
+  message: Message,
+  con: DataSource,
+  logger: FastifyBaseLogger,
+) => Promise<NotificationHandlerReturn>;
+
 export interface NotificationWorker {
   subscription: string;
+  handler: NotificationWorkerHandler;
+  maxMessages?: number;
+}
+
+interface TypedNotificationWorkerProps<
+  T extends keyof PubSubSchema,
+  D extends PubSubSchema[T],
+> {
+  subscription: T;
   handler: (
-    message: Message,
+    data: D,
     con: DataSource,
     logger: FastifyBaseLogger,
   ) => Promise<NotificationHandlerReturn>;
-  maxMessages?: number;
 }
+
+interface GenerateTypeWorkerResult<T extends keyof PubSubSchema> {
+  subscription: T;
+  handler: NotificationWorkerHandler;
+}
+
+export const generateTypedNotificationWorker = <
+  T extends keyof PubSubSchema,
+  D extends PubSubSchema[T],
+>({
+  subscription,
```

Comment:
@sshanzel basically here subscription and topic names differ so you have to at least have two params topic and subscription or refactor to use `TypedWorker` type